packages/db/migrations/custom/0002_app_widget_show_description_tooltip_to_display_mode.ts

```diff
@@ -0,0 +1,39 @@
+import SuperJSON from "superjson";
+
+import type { Database } from "../..";
+import { items } from "../../schema";
+
+/**
+ * To support showing the description in the widget itself we replaced
+ * the tooltip show option with display mode.
+ */
+export async function migrateAppWidgetShowDescriptionTooltipToDisplayModeAsync(db: Database) {
+  const existingAppItems = await db.query.items.findMany({
+    where: (table, { eq }) => eq(table.kind, "app"),
+  });
+
+  const itemsToUpdate = existingAppItems
+    .map((item) => ({
+      id: item.id,
+      options: SuperJSON.parse<{ showDescriptionTooltip?: boolean }>(item.options),
+    }))
+    .filter((item) => item.options.showDescriptionTooltip !== undefined);
+
+  console.log(
+    `Migrating app items with showDescriptionTooltip to descriptionDisplayMode count=${itemsToUpdate.length}`,
+  );
+
+  await Promise.all(
+    itemsToUpdate.map(async (item) => {
+      const { showDescriptionTooltip, ...options } = item.options;
+      await db.update(items).set({
+        options: SuperJSON.stringify({
+          ...options,
+          descriptionDisplayMode: showDescriptionTooltip ? "tooltip" : "hidden",
+        }),
+      });
```

Comment:
This would have been quite bad, I forgot to add a where clause 😂

Reply:
Fixed with [3d0a9eb](https://github.com/homarr-labs/homarr/pull/3876/commits/3d0a9eb9bac8e74bd72d48bc6eab7cf1c9405dc2)